src/main/scala/se/nimsa/sbx/storage/StorageService.scala

```diff
@@ -73,26 +77,38 @@ trait StorageService {
           }.getOrElse(ImageInformation(0, 0, 0, 0))
       }
 
-  def readPngImageData(imageId: Long, frameNumber: Int, windowMin: Int, windowMax: Int, imageHeight: Int)(implicit materializer: Materializer): Array[Byte] = {
+  def readPngImageData(imageId: Long, frameNumber: Int, windowMin: Int, windowMax: Int, imageHeight: Int)(implicit materializer: Materializer): Future[Array[Byte]] = {
     val source = fileSource(imageId)
     // dcm4che does not support viewing of deflated data, cf. Github issue #42
     // As a workaround, do streaming inflate and mapping of transfer syntax
     val inflatedSource = DicomStreamOps.inflatedSource(source)
     val is = inflatedSource.runWith(StreamConverters.asInputStream())
     val iis = ImageIO.createImageInputStream(is)
+
+
+    val imageReader = ImageIO.getImageReadersByFormatName("DICOM").next
+    imageReader.setInput(iis)
+    val param = imageReader.getDefaultReadParam.asInstanceOf[DicomImageReadParam]
+    if (windowMin < windowMax) {
+      param.setWindowCenter((windowMax - windowMin) / 2)
+      param.setWindowWidth(windowMax - windowMin)
+    }
+
+    // imageReader.read is blocking, needs to be run outside streams execution context
+    val blockingIoContext = ExecutionContext.fromExecutorService(Executors.newFixedThreadPool(2))
```

Comment:
Should this be part of a wider overhaul where we think about blocking IO and what thread pools are used to manage such code? Maybe we could use one thread pool for all such ops. I'm not sure it is a great idea to create a new pool for each call to this method. If we leave it like this we also have to shut the pool down when done, no?